src/audio/aria/aria_hifi5.c

```diff
@@ -0,0 +1,188 @@
+// SPDX-License-Identifier: BSD-3-Clause
+//
+// Copyright(c) 2024 Intel Corporation. All rights reserved.
+
+#include "aria.h"
+
+#ifdef ARIA_HIFI5
+#include <xtensa/config/defs.h>
+#include <xtensa/tie/xt_hifi5.h>
+
+/**
+ * \brief Aria gain index mapping table
+ */
+static const int32_t index_tab[] = {
+		0,    1,    2,    3,
+		4,    5,    6,    7,
+		8,    9,    0,    1,
+		2,    3,    4,    5,
+		6,    7,    8,    9,
+		0,    1,    2,    3
+};
```

Comment:
This looks like modulo 10 for 0 to 23 index. Likely not possible but if gains gains table size ARIA_MAX_GAIN_STATES  could be 16 or other 2^N value this could be bit-and instead of modulo. But it's out-of-scope of this PR to change functionality.

Reply:
> This looks like modulo 10 for 0 to 23 index. Likely not possible while keeping the algorithm the same but if gains gains table size ARIA_MAX_GAIN_STATES could be 16 or other 2^N value this could be bit-and instead of modulo. But it's out-of-scope of this PR to change functionality.